Automattic/wp-calypso
# Domains: Update Domain Management contact information section

#### Changes proposed in this Pull Request
This PR creates the new redesigned contact information component, shown in the domain settings page for registered and connected domains. This is part of the domain management pages redesign project detailed in pcYYhz-m2-p2.

#### Preview
![image](https://user-images.githubusercontent.com/18705930/146291410-f8e1d704-99c9-43fc-80c2-873e2fd415eb.png)
![image](https://user-images.githubusercontent.com/18705930/146291532-d9addfe0-babf-4842-9323-bd4906db85e9.png)
![image](https://user-images.githubusercontent.com/18705930/146294879-dc6e592c-d2d7-4b89-bcba-7234180c4901.png)


#### Testing instructions
- Build this branch locally or open the live Calypso link
- If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag
- Go to "Upgrades > Domains"
- Select one of your registered or mapped domains
- Ensure that:
  - The contact information card in the domain settings page looks like the screenshot above;
  - The privacy protection toggle works like in the old contact information page;
  - The privacy check (whether privacy is enabled or not) works the same as in the old contact information page;
  - The non-owner notice in the screenshot above is displayed when you're not the domain owner;
  - The correct information is displayed;
  - The links work correctly;
  - The "Edit" button takes you to the edit contact information page;
  - The "Manage consent" button takes you to the manage consent for personal data use page;
  - Also check that the mobile view looks correct.
  
  In short, make sure it matches the redesigned version in hlVh2q24ad6MCwlwNnE9PQ-fi-600%3A54818.
